tests/test_formats.py

```diff
@@ -725,15 +743,13 @@ def test_copy_new_uuid(self):
         self.assertNotEqual(copy.uuid, data.uuid)
         self.assertTrue(copy.data_equal(data))
 
-    @unittest.skipIf(sys.platform == "win32",
-                     "windows simultaneous file permissions issue")
     def test_copy_update_sites_inference(self):
         with formats.SampleData() as data:
             for j in range(4):
                 data.add_site(position=j, alleles=["0", "1"], genotypes=[0, 1, 1, 0])
         self.assertEqual(list(data.sites_inference), [True, True, True, True])
 
-        with tempfile.TemporaryDirectory(prefix="tsinf_format_test") as tempdir:
+        with temporary_directory(prefix="tsinf_format_test") as tempdir:
             filename = os.path.join(tempdir, "samples.tmp")
             for copy_path in [None, filename]:
                 copy = data.copy(path=copy_path)
```

Comment:
I don't think contextmanager stuff this is necessary --- we're just being sloppy about closing files. Isn't the ``copy`` still open here at the end?

Reply:
Thanks jerome. Yes, I realised that the problem is probably simply that we are not closing the SampleData file.

I'm assuming that the way we do that is simply to call [sample_data.close()](https://github.com/tskit-dev/tsinfer/blob/7442f6ff5b98497061d9729a40b953c9bc41910c/tsinfer/formats.py#L336) - I'll try testing this.

Reply:
Or use the context manager for the SampleData file --- that's what it's for.

Reply:
Ah, strike that, there may not be a context manager for the copy method.

Reply:
Yep, right. I'll go through the other tests that are commented out and see if I can use a context manager for those, though.